cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -193,10 +193,20 @@ private void addWorkflowNodeState(Id.Program program, String pid, Map<String, St
 
     WorkflowNodeStateDetail nodeStateDetail
       = new WorkflowNodeStateDetail(workflowNodeId, ProgramRunStatus.toNodeStatus(status), pid,
-                                    failureCause == null ? null
-                                      : new DefaultThrowable(Throwables.getRootCause(failureCause)));
+                                    failureCause == null ? null : new DefaultThrowable(failureCause));
 
     write(key, nodeStateDetail);
+
+    // Get the run record of the Workflow which started this program
+    key = getWorkflowRunRecordKey(workflowRunId.getParent().toId(), workflowRunId.getRun());
+
+    RunRecordMeta record = get(key, RunRecordMeta.class);
```

Comment:
There is a unit test to catch this. However when we fix JIRA - https://issues.cask.co/browse/CDAP-5458 we will need upgrade step for Workflow run records.